app/account/views.py

```diff
@@ -250,3 +250,26 @@ def unconfirmed():
     if current_user.is_anonymous() or current_user.confirmed:
         return redirect(url_for('main.index'))
     return render_template('account/unconfirmed.html')
+
+
+@account.route('/profile/<int:user_id>')
+@login_required
+def profile(user_id):
+    """Display a user's profile"""
+    user = User.query.get(int(user_id))
+    if user is None:
+        return render_template('errors/404.html')
```

Comment:
this line can be changed to `abort(404)`, http://flask.pocoo.org/docs/0.10/quickstart/#redirects-and-errors